cf_xarray/accessor.py

```diff
@@ -2633,6 +2633,54 @@ def grid_mapping_names(self) -> dict[str, list[str]]:
                 results[v].append(k)
         return results
 
+    def assign(self, **standard_variables: Any) -> Dataset:
+        """Assign new variables by standard_name.
+
+        Parameters
+        ----------
+        standard_variables : mapping
+            A mapping from netCDF Climate and Forecast compliant standard names
+            (http://cfconventions.org/standard-names.html), to the new values.
+            The new values are passed to ``:meth:Dataset.assign``.
+
+        Returns
+        -------
+        dataset : Dataset
+            A new dataset with the new variables added, replacing any existing
+            variable with the same standard name. New variables are labelled
+            according to their input (short) names, or standard names if short
+            names are missing, with added trailing underscores in the event of
+            duplicates.
+
+        See Also
+        --------
+        DataSet.assign
+        """
+
+        # initalize dictionary mapping short names to the new variables
+        variables = {}
+
+        # for each standard name and value pair
+        for standard_name, values in standard_variables.items():
+            # default to using existing short name or standard name
+            name = getattr(values, "name", standard_name)
```

Comment:
Something like 
https://github.com/xarray-contrib/cf-xarray/blob/52160a51c66edbec84f1412890887a7874dc2d25/cf_xarray/accessor.py#L2473

should be used here. So

```
mapped_name, = apply_mapper(_get_all, self._obj, standard_name, error=False, default=[standard_name])
```
 will do the translation to actual name in the dataset, if possible. Use this to create a new dictionary that you can then pass to `self._obj.assign`

Reply:
Thank you for the feedback, I will be looking at this next week.